test/Microsoft.Web.LibraryManager.Vsix.Test/UI/Models/LibraryIdViewModelTests.cs

```diff
@@ -51,18 +54,49 @@ public async Task FilterCompletions_SearchTextEndsWithAt_ReturnAllMatchingComple
         {
             var testObj = new LibraryIdViewModel(GetTestSearchService(), "test@");
 
-            CompletionSet result = await testObj.GetCompletionSetAsync(caretIndex: 0);
+            CompletionSet result = await testObj.GetCompletionSetAsync(caretIndex: 5);
 
             // the version segment is "", which matches all versions
             Assert.AreEqual(3, result.Completions.Count());
         }
 
+        [TestMethod]
+        public async Task FilterCompletions_ScopedPackageWithoutTrailingAt_ReturnAllMatchingCompletions()
+        {
+            var testObj = new LibraryIdViewModel(GetTestSearchService(), "@types/node");
+
+            CompletionSet result = await testObj.GetCompletionSetAsync(caretIndex: 11);
+
+            Assert.AreEqual(3, result.Completions.Count());
+        }
+
+        [TestMethod]
+        public async Task FilterCompletions_ScopedPackageEndsWithAt_ReturnAllMatchingCompletions()
+        {
+            var testObj = new LibraryIdViewModel(GetTestSearchService(), "@types/node@");
+
+            CompletionSet result = await testObj.GetCompletionSetAsync(caretIndex: 12);
+
+            // the version segment is "", which matches all versions
+            Assert.AreEqual(3, result.Completions.Count());
+        }
+
+        [TestMethod]
+        public async Task FilterCompletions_ScopedPackageEndsWithAfterAt_ReturnAllMatchingCompletions()
+        {
+            var testObj = new LibraryIdViewModel(GetTestSearchService(), "@types/node@2");
+
+            CompletionSet result = await testObj.GetCompletionSetAsync(caretIndex: 13);
+
+            Assert.AreEqual(1, result.Completions.Count());
```

Comment:
Does this filter differently?  The unscoped case expects 2 results when the version string is "2".